### PR TITLE
[Gutenberg] Merge post 1.72.0 changes

### PR DESCRIPTION
**Pull request by jhnstn**
Merge post Gutenberg 1.72.0 release changes:

- https://github.com/wordpress-mobile/WordPress-iOS/pull/18137

To test:

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -61,7 +61,7 @@ PODS:
     - AppAuth/Core (~> 1.4)
     - GTMSessionFetcher/Core (~> 1.5)
   - GTMSessionFetcher/Core (1.7.0)
-  - Gutenberg (1.72.0):
+  - Gutenberg (1.71.3):
```
Comment (fluiddot) [inline Podfile.lock]: The Gutenberg version is being downgraded due to the fact that tag `v1.73.0-alpha1` was created while version `1.72.0` was being released (i.e. before pushing tag `v1.72.0`). This makes me think that we should create a new tag `v1.73.0-alpha2` to include both the change from `v1.73.0-alpha1` and `v1.72.0`.

Comment (jhnstn) [reply Podfile.lock]: Thanks @fluiddot , I pushed up the new alpha release here https://github.com/wordpress-mobile/gutenberg-mobile/pull/4682

**Comment by jhnstn** [comment]
@fluiddot moving this to draft till we resolve the issue of the tag in GBM not being on trunk yet. Might end up moving the tag so just want to make sure everything there looks good before approving this PR.

**Comment by fluiddot** [comment]
@jhnstn looks like the PR was automatically closed when https://github.com/wordpress-mobile/gutenberg-mobile/pull/4682 was merged.

**Comment by jhnstn** [comment]
Hmm, not sure I like that auto close feature

File: Podfile [hunk 2]
```diff
@@ -169,7 +169,7 @@ abstract_target 'Apps' do
     ## Gutenberg (React Native)
     ## =====================
     ##
-    gutenberg :tag => 'v1.72.0'
+    gutenberg :tag => 'v1.73.0-alpha2'
```
Comment (fluiddot) [inline Podfile:172]: @jhnstn Both alpha versions `v1.73.0-alpha2` and `v1.73.0-alpha3` point to the same commit in GB-mobile, so we could use any of them on iOS. Let me know whether you rather leaving the current tag or use the newer one.

Screenshot 2022-03-23 at 09 36 36

**Review by fluiddot — APPROVED** [review]

File: Podfile (hunk 2 above)
Comment (jhnstn) [reply Podfile:172]: I'm fine leaving this at `alpha2` for now. We can bump that in a different PR if we deem that necessary.

**Pull request merged**